client/lib/wp/sync-handler/cache-invalidation.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * External dependencies
+ */
+import moment from 'moment';
+import debugFactory from 'debug';
+
+/**
+ * Internal dependencies
+ */
+import warn from 'lib/warn';
+import { getLocalForage } from 'lib/localforage';
+
+/**
+ * Module variables
+ */
+const localforage = getLocalForage();
+const debug = debugFactory( 'calypso:sync-handler:cache' );
+const RECORDS_LIST_KEY = 'records-list';
+// set record lifetime to 2 days
+const LIFETIME = 1000 * 60 * 60 * 24 * 2;
+
+export class CacheInvalidation {
+	getAll( fn = () => {} ) {
```

Comment:
I prefer use array functions instead of an external library but it's up to us. Maybe we should write some lines about how/when use lodash or an external library in general. Anyway you're commenting in a old revision of the code.